### PR TITLE
SONARPY-751 Rule S5886: Function return types should be consistent with their type hint

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**

### CODE REVIEW

File: python-checks/src/test/resources/checks/functionReturnType.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  def my_int_returns_none() -> int:
+    return  # Noncompliant
+
+  def my_int_union_type_ok(cond) -> int:
+    if cond:
+      x = 42
+    else:
+      x = "hello"
+    return x  # OK
+
```
Comment (andrea-guarino-sonarsource) [inline python-checks/src/test/resources/checks/functionReturnType.py:109]: I think we could add a test case:
```python
  def foo(cond) -> int:
    if cond:
      return 42
    else:
      return "hello"
```

Comment (guillaume-dequenne-sonarsource) [reply python-checks/src/test/resources/checks/functionReturnType.py:109]: There is a test case similar to this one on line 94-98, which does flag the `str` return as `Noncompliant`.

Comment (andrea-guarino-sonarsource) [reply python-checks/src/test/resources/checks/functionReturnType.py:109]: ops sorry I missed it

File: python-frontend/src/main/java/org/sonar/python/types/InferredTypes.java [hunk 2]
```diff
@@ -149,23 +150,32 @@ private static DeclaredType declaredTypeFromTypeAnnotation(Expression expression
...
+    }
     return null;
   }
 
   private static InferredType runtimeTypefromTypeAnnotation(Expression expression, Map<String, Symbol> builtinSymbols) {
     if (expression.is(Kind.NAME) && !((Name) expression).name().equals("Any")) {
       Symbol symbol = ((Name) expression).symbol();
-      if (symbol != null && "typing.Text".equals(symbol.fullyQualifiedName())) {
-        return InferredTypes.runtimeType(builtinSymbols.get("str"));
+      if (symbol != null ) {
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/types/InferredTypes.java]: We still branch depending on `symbol.fullyQualifiedName()` right after so I think we still need the null check.

**Pull request merged**